### ISSUE TITLE
Feature: Android SDK support

### ISSUE THREAD

**Issue by xtovski** [sdkman/sdkman-cli#1111]
Writing here as your Slack workspace doesn't work. (BTW, I guess Discord would be much better for OSS projects like this)

- [X] I have read the [CONTRIBUTING guidelines](CONTRIBUTING.md)
- [ ] I've had a conversation on the [#cli-development](https://sdkman.slack.com/app_redirect?channel=cli-development) Slack channel about this request.

**Feature request**
What about adding Google Android SDKs to SdkMan? We are using quite simple shell scripts to install Android SDKs now, but I think SdkMan would benefit from having Android SDK supported. If community has no objections we could help with PR for this feature.

**Comment by amochkin** [comment]
I wonder if there is any guide how to add new SDK to SdkMan? From what I could discover we will need to modify:
- https://github.com/sdkman/sdkman-db-migrations
- https://github.com/sdkman/sdkman-candidates

**Comment by xtovski** [comment]
What about this one?
https://github.com/sdkman/sdkman-cli/wiki/Vendor-onboarding-process

**Comment by helpermethod** [comment]
Hi!

Don't you need to accept a license agreement when downloading the Android SDK?

**Comment by xtovski** [comment]
I guess even CLI tools would be great already.
Android SDKs are managed by `sdkmanager` and it only needs to have ANDROID_SDK_ROOT set.
If SDKMAN! can do that it would make life much easier for a lot of people.

**Comment by helpermethod** [comment]
@xtovski So, in theory it would be possible to provide the CLI Tools via SDKMAN! but it's  there no way for SDKMAN! to set the ANDROID_SDK_ROOT environment variable.

Not sure how helpful that would be 😅.

**Comment by helpermethod** [comment]
The problem is that SDKMAN! just takes the candidate name, uppercases it and then appends `_HOME` to it

```
local candidate_home_var="$(echo ${candidate_name} | tr '[:lower:]' '[:upper:]')_HOME"
```

**Comment by xtovski** [comment]
Understood. I am researching if setting env var `ANDROID_HOME` will be sufficient.

**Comment by marc0der** [comment]
We don't do any SDK-specific hacks in the code. The `_HOME` environment variables are strictly not even needed but are just there for convenience and as a loose convention among Java/JDK SDKs. We had the same questions when we brought Maven onto SDKMAN (it requires(ed) an `M2_HOME`, which we simply didn't accommodate. If `ANDROID_HOME` doesn't do it for you, just set that env var manually.

**Comment by xtovski** [comment]
> But first, we should probably decide if it is viable to even have the Android tools on SDKMAN.

I agree with that. Actually managing whole set of Android SDKs makes no sense since it is a thing-in-itself. But bootstrapping Android SDK by installing Android CLI tools along with Android sdkmanager still makes some sense. Right now developers are required to download and unpack it manually.

**Comment by helpermethod** [comment]
@marc0der The ZIP file structure looks good, and unpacking the `sdkmanager` runs fine.

Apart from setting the env variable manually I think there is still some benefit in managing it via SDKMAN!:

* no manual, platform-specific `.zip` download and unpacking
* Java is required anyway, which SDKMAN! already handles
* upgrades become much easier

The only thing I'm a little bit worried about is that a `Terms And Conditions` pops up when I try to download the `sdkmananger` binary. Though I can still grab the URL and download the ZIP just fine from the commandline (the Homebrew formula also just downloads the binary).

**Comment by xtovski** [comment]
> The only thing I'm a little bit worried about is that a `Terms And Conditions` pops up when I try to download the `sdkmananger` binary. Though I can still grab the URL and download the ZIP just fine from the commandline (the Homebrew formula also just downloads the binary).

Yeah, you are correct. There is a license modal requiring a user to accept license terms before downloading.
Do you think we should get a clarification from Google if we are allowed to download the tools using direct link?

**Comment by marc0der** [comment]
Those T&C's are a deal breaker for me. Right now we still have the pre-hook which could handle it, but we are about to remove that functionality all together. I don't think we should be hosting any SDKs with prohibitive licencing.
